doc/development/plugins.rst

```diff
@@ -1,395 +1,134 @@
-:math:`q_0` is the most significant bit.
-
-And thats it! The device has inherited :meth:`~.QubitDevice.expval`, :meth:`~.QubitDevice.var`,
-and :meth:`~.QubitDevice.sample` methods, that accepts an observable (or tensor product of
-observables) and returns the corresponding measurement statistic.
-
-
-
-:html:`<div class="caution admonition" id="aside1"><a data-toggle="collapse" data-parent="#aside1" href="#content1" class="collapsed"><p class="first admonition-title">Advanced execution control (click to expand) <i class="fas fa-chevron-circle-down"></i></p></a><div id="content1" class="collapse" data-parent="#aside1" style="height: 0px;">`
-
-Additional flexibility is sometimes required for interfacing with more
-complicated frameworks.
-
-When PennyLane needs to evaluate a QNode, it accesses the :meth:`~.QubitDevice.execute` method of
-your plugin which, by default, performs the following process:
-
-.. code-block:: python
-
-    self.check_validity(circuit.operations, circuit.observables)
-
-    # apply all circuit operations
-    self.apply(circuit.operations, rotations=circuit.diagonalizing_gates)
-
-    # generate computational basis samples
-    if self.shots is not None or circuit.is_sampled:
-        self._samples = self.generate_samples()
-
-    # compute the required statistics
-    results = self.statistics(circuit)
-
-    return self._asarray(results)
-
-where
-
-* ``circuit`` is a :class:`~.CircuitGraph` object
-
-* :attr:`circuit.operations <pennylane.CircuitGraph.operations>` are the user-provided
-  operations to be executed
-
-* :attr:`circuit.observables <pennylane.CircuitGraph.observables>` are the user-provided
-  observables to be measured
-
-* :attr:`circuit.diagonalizing_gates <pennylane.CircuitGraph.diagonalizing_gates>` are the
-  gates that rotate the circuit prior to measurement so that computational basis
-  measurements are performed in the eigenbasis of the requested observables
-
-* :meth:`.QubitDevice.statistics` returns the results of :meth:`.QubitDevice.expval`,
-  :meth:`~.QubitDevice.var`, or :meth:`~.QubitDevice.sample` depending on the type
-  of observable.
-
-In advanced cases, the :meth:`.QubitDevice.execute` method, as well as
-:meth:`.QubitDevice.statistics`, may be overwritten directly.
-This provides full flexibility for handling the device execution yourself. However,
-this may have unintended side-effects and is not recommended.
-
-
-:html:`</div></div>`
-
-
-Wire handling
--------------
-
-PennyLane uses the :class:`~.wires.Wires` class for the internal representation of wires. :class:`~.wires.Wires`
-inherits from Python's ``Sequence``, and represents an ordered set of unique wire labels.
-Indexing a ``Wires`` instance will return another ``Wires`` instance of length one.
-The ``labels`` attribute stores a tuple of the wire labels.
+        def execute(self, circuits: "QuantumTape_or_Batch", execution_config: "ExecutionConfig" = DefaultExecutionConfig):
```

Comment:
> QuantumTape_or_Batch

Is this in place of a `Union`?

Reply:
I probably should just right out `Union[QuantumTape, Sequence[QuantumTape]]`.